docs/Release.md

```diff
@@ -0,0 +1,112 @@
+# Amazon Elastic File System (EFS) CSI driver Release Process
+
+## Choose the release version and release branch
+
+1. Find the latest release:
+   https://github.com/kubernetes-sigs/aws-efs-csi-driver/releases. For example,
+   `v1.4.8`. (Ignore helm releases prefixed by `helm-chart` like
+   `helm-chart-aws-efs-csi-driver-2.3.6`).
+2. Increment the version according to semantic versioning https://semver.org/.
+   For example, for a release that only contains bug fixes or an updated Amazon
+   Linux 2 base image, `v1.4.9`.
```

Comment:
I would get rid of the example, the version at the time someone is reading this likely won't be 1.4.8